src/macaron/__main__.py

```diff
@@ -208,7 +238,15 @@ def main(argv: list[str] | None = None) -> None:
 
     # Use Macaron to analyze one single repository.
     single_analyze_parser = sub_parser.add_parser(name="analyze")
-    group = single_analyze_parser.add_mutually_exclusive_group(required=True)
+
+    # We set this group, which makes the usage of --config-path and --repo-path to be mutual exclusive, to optional
+    # so that the user could provide the --package-url by itself while persisting the current behavior of Macaron.
+    # Note that if the user provide both --purl and --config-path, we will still raise error, which is handled within
+    # the ``analyze_slsa_levels_single`` method.
+    # When we remove the --config-path option is moved, we can remove this group and instead adding all relevant
+    # options in the analyze command through ``single_analyze_parser``.
```

Comment:
```suggestion
    # We make the mutually exclusive usage of --config-path and --repo-path optional
    # so that the user can provide the --package-url separately while keeping the current behavior of Macaron.
    # Note that if the user provides both --package-url and --config-path, we will still raise an error, which is handled within
    # the ``analyze_slsa_levels_single`` method.
    # When we remove the --config-path option, we can remove this group and instead add all relevant
    # options in the analyze command through ``single_analyze_parser``.
```

Reply:
Fixed in https://github.com/oracle/macaron/pull/401/commits/f66b0be2bf3c2556eed5885315d39cd66b792306